src/virtualkeyboardanthy.cpp

```diff
@@ -24,6 +24,10 @@ AnthyKeyboard::AnthyKeyboard()
         << "path of Japanese keyboard layout file: " << jsonPath;
     loader_ = new KeyboardLayout(jsonPath);
 }
+AnthyKeyboard::~AnthyKeyboard() {
+    FCITX_KEYBOARD_LAYOUT() << "~AnthyKeyboard()";
+    delete loader_;
```

Comment:
Should we use `unique_ptr` for `loader_`?

Reply:
Changed to use unique_ptr.

Reply:
Even though loader_ was handled by unique_ptr, it is not so effective to reduce leaks.
As already pointed out, `keys` must be correctly destroyed.

Reply:
but delete against up-casted instance causes undefined behavior.

Reply:
> but delete against up-casted instance causes undefined behavior.

They will be correctly deleted when they inherits parent class with `public` and define public destructor. It seems that all child classes satisfy these conditions, so I believe they can delete safely.